backbone.js

```diff
@@ -1088,8 +1088,8 @@
     },
 
     // Define how to uniquely identify models in the collection.
-    modelId: function(attrs) {
-      return attrs[this.model.prototype.idAttribute || 'id'];
+    modelId: function(attrs, model) {
+      return attrs[(model && model.idAttribute) || this.model.prototype.idAttribute || 'id'];
```

Comment:
Instead of passing the entire model, you could pass the idAttribute of the model.

Then you could more easily implement a get that would work with your example.


```
collection.get({companyId:1234}, 'companyId');
collection.get({customerId:1234}, 'customerId');
```

```
 get: function(obj, idAttribute) {
      if (obj == null) return void 0;
      var id = this.modelId(this._isModel(obj) ? obj.attributes : obj, idAttribute);
      return this._byId[obj] || this._byId[id] || this._byId[obj.cid];
    },
```

```
modelId: function (attrs, idAttribute) {
    return attrs[idAttribute || this.model.prototype.idAttribute || 'id'];
},
```